src/core/federation/merge.rs

```diff
@@ -0,0 +1,21 @@
+use crate::core::merge_right::MergeRight;
+use crate::core::primitive::Primitive;
+use crate::core::valid::Valid;
+
+pub trait FederatedMerge: Sized {
+    fn federated_merge(self, other: Self) -> Valid<Self, String>;
+}
+
+pub trait FederatedMergeIntersection: Sized {
+    fn federated_merge_intersection(self, other: Self) -> Valid<Self, String>;
+}
+
+pub trait FederatedMergeUnion: Sized {
+    fn federated_merge_union(self, other: Self) -> Valid<Self, String>;
+}
```

Comment:
Don't think we need three traits. The following might be more concise and meaningful to work with.

```rust
pub trait FederatedMerge: Sized {
    fn federated_merge_right(self, other: Self) -> Valid<Self, String>;
    fn federated_merge_intersection(self, other: Self) -> Valid<Self, String>;
    fn federated_merge_union(self, other: Self) -> Valid<Self, String>;
}
```

Reply:
no, it's not how it meant to be used. Different config entities could implement different strategies of merging and not every one of them how it's suggested. The traits are used to provide type checking of available strategies and reuse code for different entities